docs/contributing/development-guide.md

```diff
@@ -0,0 +1,345 @@
+any chances of your data being unintentionally modified. Once you have the app setup
+and a valid access_token for an account with at least one activity, follow the steps
+below.
+
+1. Rename the file `stravalib/stravalib/tests/test.ini-example` to `test.ini`
+2. Add your API token to the file by replacing:
+
+```bash
+access_token = xxxxxxxxxxxxxxxx
+```
+with: 
+
+```bash
+access_token = your-authentication-token-value-here
+```
+
+NOTE: this token needs to have write access to your account. 
+We recommend that you create
+a dummy account to ensure you aren't modifying your actual account data.
+
+3. Add a single activity id to your dummy account using stravalib:
+
+```bash
+activity_id = a-valid-activity-id-here
+```
+
+You are now ready to run the test suite. To run tests on python 3.x run:
+
+```bash
+$ pytest
+```
+
+For integration tests that should be run independently from Strava, there's a pytest
+fixture :func:`~stravalib.tests.integration.conftest.mock_strava_api`
+that is based on :class:`responses.RequestsMock`.
+It prevents requests being made to the actual Strava API and instead registers responses
+that are based on examples from the published Strava API documentation. Example usages of
+this fixture can be found in the :mod:`stravalib.tests.integration.test_client` module.
+
+### Test code coverage 
+We use [pytest-cov](https://pytest-cov.readthedocs.io/en/latest/) to calculate 
+test coverage. When you run `make test` pytest-cov will provide you with coverage 
+outputs locally. You can ignore the returned values for any files in the `test`
+directory.  
+
+Example output from `make test`:
+
+```bash
+pytest --cov stravalib stravalib/tests/unit stravalib/tests/integration 
+=================================================== test session starts ===================================================
+platform darwin -- Python 3.8.13, pytest-7.2.0, pluggy-1.0.0
+rootdir: /Users/leahawasser/Documents/GitHub/stravalib
+plugins: cov-4.0.0
+collected 105 items                                                                                                       
+
+stravalib/tests/unit/test_attributes.py ...............                                                             [ 14%]
+stravalib/tests/unit/test_client_utils.py .......                                                                   [ 20%]
+stravalib/tests/unit/test_limiter.py .............                                                                  [ 33%]
+stravalib/tests/unit/test_model.py .......                                                                          [ 40%]
+stravalib/tests/integration/test_client.py ...............................................................          [100%]
+
+---------- coverage: platform darwin, python 3.8.13-final-0 ----------
+Name                                                     Stmts   Miss  Cover
+----------------------------------------------------------------------------
```

Comment:
NOTE TO SELF: update this and remove tests from the file. That was the pr i merged this morning.

Reply:
ok fixed this locally will push in a minute.